src/Functions/FunctionsAES.h

```diff
@@ -409,12 +416,14 @@ template <typename Impl>
 class FunctionDecrypt : public IFunction
 {
 public:
+    explicit FunctionDecrypt(bool use_null_when_decrypt_fail_) : use_null_when_decrypt_fail(use_null_when_decrypt_fail_) { }
     static constexpr OpenSSLDetails::CompatibilityMode compatibility_mode = Impl::compatibility_mode;
     static constexpr auto name = Impl::name;
-    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionDecrypt>(); }
+    static FunctionPtr create(ContextPtr context) { return std::make_shared<FunctionDecrypt>(AESHelpers::getParamsFromContext(context)); }
```

Comment:
I still don't understand, are there any plans to use a separate function in a separate namespace `AESHelpers::getParamsFromContext` elsewhere? It seems it will only be used here, so why not get the setting right here, or in the FunctionDecrypt constructor without calling a separate function? :)

Reply:
It won't be used elsewhere. In order to call method of context, you need to include "Context.h", but this is the header file, so it will create a circular dependency among header files if you include "Context.h" here. On the other hand, we cannot move the constructor to .cpp file since this is a template. So I create this function because I can have its definition on .cpp file.
If there's a more elegant solution, pls let me know.